src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs

```diff
@@ -375,6 +375,9 @@ internal static void UpdateMenuItemIcon(this IMauiContext mauiContext, IMenuItem
 						}
 
 						menuItem.SetIcon(iconDrawable);
+
+						// Setting the title, so that tooltip text is displayed in native color
+						menuItem.SetTitle(toolBarItem?.Text ?? string.Empty);
```

Comment:
Don't we already set this in UpdateMenuItem?

Is there some code path that is removing it? Also, how does re-setting the text change the color?